### PR TITLE
show selected gift card

### PR DESCRIPTION
**Pull request by sawangupta92**

### CODE REVIEW

File: app/controllers/spree/gift_cards_controller.rb [hunk 1]
```diff
@@ -2,6 +2,7 @@ module Spree
   class GiftCardsController < Spree::StoreController
 
     def new
+      @variant = Spree::Product.find_by(slug: params[:product_id]).try(:master)
```
Comment (nishant-cyro) [inline app/controllers/spree/gift_cards_controller.rb]: @sawangupta92 - Move it to before_action.

File: app/controllers/spree/products_controller_decorator.rb [hunk 2]
```diff
@@ -5,7 +5,7 @@
   private
 
     def redirect_gift_card
-      redirect_to new_gift_card_path and return false if @product.try :is_gift_card?
+      redirect_to new_gift_card_path(product_id: @product) and return false if @product.try :is_gift_card?
```
Comment (nishant-cyro) [inline app/controllers/spree/products_controller_decorator.rb]: @sawangupta92 - why we need to pass product_id here.

File: app/overrides/spree/orders/_line_item/replace_cart_item_quantity.html.erb.deface [hunk 3]
```diff
@@ -0,0 +1,7 @@
+<!-- replace_contents '[data-hook="cart_item_quantity"]' -->
+
+<% if line_item.gift_card.present? %>
+  <%= item_form.number_field :quantity, min: 0, class: "form-control line_item_quantity", size: 5, max: 1 %>
```
Comment (nishant-cyro) [inline app/overrides/spree/orders/_line_item/replace_cart_item_quantity.html.erb.deface]: @sawangupta92 - Move max to a preference.

File: app/controllers/spree/products_controller_decorator.rb (hunk 2 above)
Comment (sawangupta92) [inline app/controllers/spree/products_controller_decorator.rb]: @nishant-cyro we need to display the product as already selected, which was selected from product listing.

**Pull request merged**